src/components/DappBrowser/hooks/useAnimatedTab.ts

```diff
@@ -58,116 +111,146 @@ export function useAnimatedTab({ tabId }: { tabId: string }) {
     return defaultBackgroundColor;
   });
 
-  const animatedTabXPosition = useDerivedValue(() =>
-    withSpring((animatedTabIndex.value % 2) * (TAB_VIEW_COLUMN_WIDTH + 20) - (TAB_VIEW_COLUMN_WIDTH + 20) / 2, SPRING_CONFIGS.slowSpring)
-  );
-
-  const animatedTabYPosition = useDerivedValue(() =>
-    withSpring(Math.floor(animatedTabIndex.value / 2) * TAB_VIEW_ROW_HEIGHT - 181, SPRING_CONFIGS.slowSpring)
-  );
-
-  const animatedWebViewBackgroundColorStyle = useAnimatedStyle(() => ({
-    backgroundColor: safeBackgroundColor.value,
-  }));
+  const animatedWebViewBackgroundColorStyle = useAnimatedStyle(() => {
+    const tabUrl = animatedTabUrls.value[tabId] || RAINBOW_HOME;
+    const isOnHomepage = tabUrl === RAINBOW_HOME;
+    return {
+      backgroundColor: isOnHomepage ? homepageBackgroundColor : safeBackgroundColor.value,
+    };
+  });
 
   const animatedWebViewStyle = useAnimatedStyle(() => {
-    const isTabBeingClosed = currentlyOpenTabIds.value.indexOf(tabId) === -1 && currentlyOpenTabIds.value.length !== 0;
-    const animatedIsActiveTab = animatedActiveTabIndex.value === animatedTabIndex.value;
-
-    const opacity = interpolate(tabViewProgress.value, [0, 100], [animatedIsActiveTab ? 1 : 0, 1], 'clamp');
-
-    const scaleDiff = 0.7 - TAB_VIEW_COLUMN_WIDTH / DEVICE_WIDTH;
-    const scale = interpolate(
-      tabViewProgress.value,
-      [0, 100],
-      [
-        animatedIsActiveTab && !isTabBeingClosed ? 1 : TAB_VIEW_COLUMN_WIDTH / DEVICE_WIDTH,
-        0.7 - scaleDiff * (isTabBeingClosed ? 1 : animatedMultipleTabsOpen.value),
-      ]
-    );
+    const tabIndex = currentlyOpenTabIds.value.indexOf(tabId);
+    const activeIndex = Math.abs(animatedActiveTabIndex.value);
+    const pendingActiveIndex = activeIndex + pendingTabSwitchOffset.value;
+    const isPendingActiveTab = pendingActiveIndex === tabIndex;
+    const animatedIsActiveTab = isPendingActiveTab || currentlyOpenTabIds.value.length === 0;
 
-    const xPositionStart = animatedIsActiveTab && !isTabBeingClosed ? 0 : animatedTabXPosition.value;
-    const xPositionEnd = (isTabBeingClosed ? 1 : animatedMultipleTabsOpen.value) * animatedTabXPosition.value;
-    const xPositionForTab = interpolate(tabViewProgress.value, [0, 100], [xPositionStart, xPositionEnd]);
+    const isRunningEnterTabViewAnimation = tabViewGestureState.value === TabViewGestureStates.DRAG_END_ENTERING;
+    const isLeftOfActiveTab = tabIndex === pendingActiveIndex - 1;
+    const isRightOfActiveTab = tabIndex === pendingActiveIndex + 1;
```

Comment:
These values are recalculated for each tab in a few places — going to create some helper values in `activeTabInfo` to reduce the redundancy.

Reply:
Added a `getTabInfo` utility instead to use the existing Shared Values directly